iib/web/status.py

```diff
@@ -0,0 +1,140 @@
+        return False, "database connection failed"
+    except sqlalchemy.exc.SQLAlchemyError as err:
+        log.error("an error occurred while trying to connect to the DB server: %s", err)
+        return False, "an error occurred"
+    finally:
+        session.close()
+
+    return True, None
+
+
+def rabbitmq_connection_check(broker_url):
+    """
+    Check if RabbitMQ connection is working.
+
+    :param str broker_url: the AMQP(S) URL to connect to RabbitMQ
+    :return: tuple (ok: bool, reason: str or None)
+    """
+    with kombu.Connection(broker_url) as connection:
+        try:
+            connection.ensure_connection(max_retries=0)
+        except kombu.exceptions.OperationalError as OperError:
+            log.exception("broker connection failed, exception: %s", OperError)
+            return False, "broker connection failed"
+        except kombu.exceptions.KombuError as err:
+            log.error("an error occurred while trying to connect to the RabbitMQ server: %s", err)
+            return False, "an error occurred"
+
+    return True, None
+
+
+def _ping_celery_workers(retries):
+    """
+    Attempt to ping worker, retry on ConnectionError.
+
+    :param int retries: number of retry/retries before returning an empty response
+    :return: dict of {hostname: reply} as returned by ping()
+    """
+    for i in range(retries + 1):
+        if i > 0:
+            time.sleep(0.25 * 2 ** (i - 1))
+        try:
+            replies = app.control.inspect().ping()
+        except ConnectionError:
+            continue
+
+        return replies or {}
+
+    return {}
+
+
+def workers_status(retries=2):
+    """
+    Ping workers, check received replies.
+
+    :param int retries: number of retry/retries on ConnectionError when pinging workers
+    :return: list of workers status information (empty if no replies)
+    """
+    replies = _ping_celery_workers(retries=retries)
+
+    # replies is a dict of {hostname: reply}, convert to a sorted list of [(hostname, reply)]
+    reply_tuples = sorted(replies.items(), key=lambda kv: kv[0])
+    workers = []
+
+    for worker_name, reply in reply_tuples:
+        # the reply is in the following format {'ok': 'pong'}.
+        worker = {"name": worker_name, "ok": "ok" in reply}
+
+        if "ok" not in reply:
+            worker["reason"] = reply.get("error", "unknown reason")
+
+        workers.append(worker)
+
+    return workers
+
+
+def status(short=False, worker_ping_retries=2):
+    """
+    Get status of IIB workers and services that IIB depends on.
+
+    Display the current version of podman, skopeo, opm.
+    :param bool short: raise an error as soon as any problem is found
+    :param int worker_ping_retries: number of retry/retries on ConnectionError when pinging workers
+    :return: dict with the following keys:
+        "services": list of status info for individual services
+        "software": list of softwares and their versions
```

Comment:
This was for previous version of this function, right?
Since I see return just "status" and "workers".